components/eventing-controller/reconciler/apirule/reconciler.go

```diff
@@ -53,12 +53,7 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 // Reconcile handles the APIRule add/update/delete.
 func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
-	apiRule := &apigatewayv1alpha1.APIRule{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: req.Namespace,
-		},
-	}
+	apiRule := &apigatewayv1alpha1.APIRule{}
```

Comment:
The the APIRule deletion scenario requires to have an APIRule instance with the basic info available (name, namespace).

```golang
// this logic is required
apiRule := &apigatewayv1alpha1.APIRule{
	ObjectMeta: metav1.ObjectMeta{
		Name:      req.Name,
		Namespace: req.Namespace,
	},
}
``` 

For example, the following functions except an APIRule instance as well to work correctly.

```golang
handleAPIRuleDelete(ctx context.Context, apiRule *apigatewayv1alpha1.APIRule)

syncSubscriptionsStatus(ctx context.Context, apiRule *apigatewayv1alpha1.APIRule, subscriptions []eventingv1alpha1.Subscription, log logr.Logger)
```